src/client/hud.cpp

```diff
@@ -862,6 +862,56 @@ void Hud::drawSelectionMesh()
 	}
 }
 
+void Hud::toggleBlockBounds()
+{
+	m_block_bounds_mode = static_cast<BlockBoundsMode>(m_block_bounds_mode + 1);
+
+	if (m_block_bounds_mode >= BLOCK_BOUNDS_MAX) {
+		m_block_bounds_mode = BLOCK_BOUNDS_OFF;
+	}
+}
+
+void Hud::drawBlockBounds()
+{
+	if (m_block_bounds_mode == BLOCK_BOUNDS_OFF) {
+		return;	
+	}
+
+	video::SMaterial old_material = driver->getMaterial2D();
+	driver->setMaterial(m_selection_material);
+
+	v3s16 pos = player->getStandingNodePos();
+
+	v3s16 blockPos(
+		floorf((float) pos.X / MAP_BLOCKSIZE),
+		floorf((float) pos.Y / MAP_BLOCKSIZE),
+		floorf((float) pos.Z / MAP_BLOCKSIZE)
+	);
+
+	v3f offset = intToFloat(client->getCamera()->getOffset(), BS);
+
+	s8 radius = 2;
+
+	v3f halfNode = v3f(BS, BS, BS) / 2.0f;
+
+	for (s8 x = -radius; x <= radius; x++)
+	for (s8 y = -radius; y <= radius; y++)
+	for (s8 z = -radius; z <= radius; z++) {
+		v3s16 blockOffset(x, y, z);
+
+		if (blockOffset.getLengthSQ() == 0 || m_block_bounds_mode == BLOCK_BOUNDS_ALL) {
```

Comment:
Even easier, if mode isn't `BLOCK_BOUNDS_ALL` just set radius to 0.

Reply:
This was actually my initial idea but I thought it would cause the loop to skip. Guess that was premature; the loop is <= not < 😁